uaclient/entitlements/base.py

```diff
@@ -444,39 +480,33 @@ def enable(
                 == CanEnableFailureReason.INACTIVE_REQUIRED_SERVICES
             ):
                 # Try to enable those services before proceeding with enable
-                req_ret, error = self._enable_required_services()
+                req_ret, error = self._enable_required_services(progress)
                 if not req_ret:
                     fail.message = error
                     return False, fail
             else:
                 # every other reason means we can't continue
                 return False, fail
 
-        msg_ops = self.messaging.get("pre_enable", [])
-        if not util.handle_message_operations(msg_ops):
-            return False, None
+        progress.emit("message_operation", self.messaging.get("pre_enable"))
 
         # TODO: Move all logic from RepoEntitlement that
         # handles the additionalPackages and APT directives
         # to the base class. The additionalSnaps handling
         # is a step on that direction
         if not self.access_only:
-            if not self.handle_required_snaps():
+            if not self.handle_required_snaps(progress):
                 return False, None
-            if not self.handle_required_packages():
+            if not self.handle_required_packages(progress):
                 return False, None
 
-        ret = self._perform_enable(silent=silent)
+        ret = self._perform_enable(progress)
         if not ret:
             return False, None
 
-        msg_ops = self.messaging.get("post_enable", [])
```

Comment:
we are not emit-ing post enable messages here, similar to pre enable?

Reply:
When we specced the api for enable, we decided that the api would return the post_enable messages. So I'm moving the handling of the out of the entitlement and into the cli/enable module